local-deploy.sh

```diff
@@ -0,0 +1,232 @@
+else
+    echo "FAIL: maven was not found on your system"
+fi
+
+if [[ "$_mvn" ]]; then
+    version=$("$_mvn" -version 1>&1 | awk -F '"' '/Apache/ {print $0}')
+    vCompare_mvn=$(to_major_version "${version}")
+    echo version: "$version"
+    if (( $(echo "$vCompare_mvn" "$minMaven" | awk '{print ($1 > $2)}') )); then
+        echo "PASS: Apache Maven version is more than 3.0"
+    else         
+        echo "WARN: Apache Maven version is less than 3.0. Please Update."
+    fi
+fi
+
+#----------------------------------------------------------------------------------
+
+echo
+echo "Validating nodejs..........................................-"
+
+if command_exists nodejs; then
+    echo "PASS: found nodejs executable in PATH"
+    _nodejs=nodejs
+else
+    echo "FAIL: nodejs was not found on your machine"
+fi
+
+if [[ "$_nodejs" ]]; then
+    version=$("$_nodejs" -v 2>&1 | tr -d 'v')
+    vCompare_node=$(to_major_version "${version}")
+    echo version: "$version"
+    if (( $(echo "$vCompare_node" "$minNode" | awk '{print ($1 > $2)}') )); then
+        echo "PASS: nodejs version is more than 4.0"
+    else         
+        echo "WARN: nodejs version is less than 4.0"
+    fi
+fi
+if command_exists npm; then
+    echo "PASS: found npm executable in PATH"
+    _npm=npm
+else
+    echo "FAIL: npm was not found on your machine"
+fi
+
+if [[ "$_npm" ]]; then
+    version=$("$_npm" -v )
+    echo version: "$version"
+    vCompare_npm=$(to_major_version "${version}")
+    if (( $(echo "$vCompare_npm" "$minNpm" | awk '{print ($1 > $2)}') )); then
+        echo "PASS: npm version is more than 3.0"
+    else         
+        echo "WARN: npm version is less than 3.0"
+    fi
+fi
+#----------------------------------------------------------------------------------
+
+
+echo
+echo "Validating scala..........................................-"
+
+if command_exists scala; then
+    echo "PASS: found scala executable in PATH"
+    _scala=scala
+else
+    echo "FAIL: scala was not found on your machine."
+fi
+
+if [[ "$_scala" ]]; then
+    version=$("$_scala" -version 2>&1 | cut -d' ' -f5)
+    echo version: "$version"
+    vCompare_scala=$(to_major_version "${version}")
+    if (("$vCompare_scala" > "$minScala")); then
+        echo "PASS: scala version is more than 2.7"
+    else         
+        echo "WARN: scala version is less than 2.7" 
+    fi
+fi
+
+#----------------------------------------------------------------------------------
+
+echo
+echo "Validating sbt..........................................-"
+
+if command_exists sbt; then
+    echo "PASS: found sbt directory in PATH"
+    echo Checking fortis project sbt version.... "$(sbt sbtVersion)"
+else
+    echo "FAIL: sbt was not found on your machine."
+fi
+
+#----------------------------------------------------------------------------------
+
+echo
+echo "Validating cassandra..........................................-"
+
+if command_exists -p cassandra; then
```

Comment:
Do we actually have a hard dependency on Cassandra? As a developer, I don't need to run a Cassandra instance on my local machine. I can point to one that's elsewhere. I'd recommend removing this, in order to defer the that choice to the developer. Perhaps a warning would be appropriate?